src/System.IO.Compression/src/System/IO/Compression/DeflateManaged/FastEncoderStatus.cs

```diff
@@ -204,24 +207,24 @@ static FastEncoderStatics()
             int code;
             for (code = 0; code < 16; code++)
             {
-                for (int n = 0; n < (1 << FastEncoderStatics.ExtraDistanceBits[code]); n++)
-                    s_distLookup[dist++] = (byte)code;
+                for (int n = 0; n < (1 << ExtraDistanceBits[code]); n++)
+                    result[dist++] = (byte)code;
             }
 
-            dist >>= 7; // from now on, all distances are divided by 128 
+            dist >>= 7; // from now on, all distances are divided by 128
 
-            for (; code < FastEncoderStatics.NumDistBaseCodes; code++)
+            for (; code < NumDistBaseCodes; code++)
             {
-                for (int n = 0; n < (1 << (FastEncoderStatics.ExtraDistanceBits[code] - 7)); n++)
-                    s_distLookup[256 + dist++] = (byte)code;
+                for (int n = 0; n < (1 << (ExtraDistanceBits[code] - 7)); n++)
+                    result[256 + dist++] = (byte)code;
```

Comment:
Yes. Instead of this being a static constructor that initializes `s_distLookup`, I've converted it to a method that creates and returns the array, called when initializing the static `s_distLookup` field.